test/framework/cluster_proxy.go

```diff
@@ -141,6 +141,13 @@ func WithMachineLogCollector(logCollector ClusterLogCollector) Option {
 	}
 }
 
+// WithRESTConfigModifier allows to modify the rest config in GetRESTConfig.
```

Comment:
nit 
I would like a short explanation about why we need this.
e.g. 
```suggestion
// WithRESTConfigModifier allows to modify the rest config in GetRESTConfig.
// Using this function it is possible to create ClusterProxy that can work with workload clusters hosted in places 
// not directly accessible from the machine where we run the E2E tests, eg. inside kind.
```